producer_test.go

```diff
@@ -194,6 +194,22 @@ func TestProducer(t *testing.T) {
 	}
 }
 
+func TestProducerNoPuts(t *testing.T) {
```

Comment:
is this test necessary to keep around or was it more just for trying it out locally?

Reply:
I think it's necessary. I added it to make sure I wasn't causing deadlock when all the connections were idle. Will be good to keep around to prevent future changes causing deadlock I think.